src/lib.rs

```diff
@@ -199,4 +212,80 @@ impl Contract {
         self.assert_deposit_whitelist(&env::predecessor_account_id());
         self.deposit_whitelist.remove(account_id.as_ref());
     }
+
+    pub fn create_draft_group(&mut self) -> DraftGroupIndex {
+        self.assert_deposit_whitelist(&env::predecessor_account_id());
+
+        let index: DraftGroupIndex = self.draft_groups.len() as _;
+        self.draft_groups.push(&DraftGroup::new());
+
+        index
+    }
+
+    pub fn create_draft(&mut self, draft: Draft) -> DraftIndex {
+        self.assert_deposit_whitelist(&env::predecessor_account_id());
+        draft.assert_new_valid(&env::predecessor_account_id().try_into().unwrap());
+        let mut draft_group = self
+            .draft_groups
+            .get(draft.draft_group_id as _)
+            .expect("draft group not found");
+        draft_group.assert_can_add_draft();
+
+        let index = self.next_draft_id;
+        self.next_draft_id += 1;
+        assert!(self.drafts.insert(&index, &draft).is_none(), "Invariant");
+
+        draft_group.total_amount += draft.total_balance();
+        draft_group.draft_indices.insert(index);
+        self.draft_groups
+            .replace(draft.draft_group_id as _, &draft_group);
+
+        index
+    }
+
+    pub fn create_drafts(&mut self, drafts: Vec<Draft>) -> Vec<DraftIndex> {
+        drafts
+            .into_iter()
+            .map(|draft| self.create_draft(draft))
+            .collect()
+    }
+
+    pub fn convert_draft(&mut self, draft_id: DraftIndex) -> LockupIndex {
+        let mut draft = self.drafts.remove(&draft_id).expect("draft not found");
+        let mut draft_group = self
+            .draft_groups
+            .get(draft.draft_group_id as _)
+            .expect("draft group not found");
+        draft_group.assert_can_convert();
+
+        // remove draft from indices and total amount
+        assert!(draft_group.draft_indices.remove(&draft_id), "Invariant");
+        let amount = draft.total_balance();
+        assert!(draft_group.total_amount >= amount, "Invariant");
+        draft_group.total_amount -= amount;
+        self.draft_groups
+            .replace(draft.draft_group_id as _, &draft_group);
+
+        // update payer_id
+        if let Some(mut termination_config) = draft.lockup.termination_config {
+            termination_config.payer_id = draft_group.payer_id.unwrap();
+            draft.lockup.termination_config = Some(termination_config);
+        }
```

Comment:
Do I miss something or you could just borrow mutably here?

```suggestion
        if let Some(&mut termination_config) = draft.lockup.termination_config {
            termination_config.payer_id = draft_group.payer_id.unwrap();
        }
```